src/store/plugins/fungibleTokens.js

```diff
@@ -158,6 +159,29 @@ export default (store) => {
           decodedResult !== undefined ? 'change_allowance' : 'create_allowance'
         ](activeNetwork.tipContractV2.replace('ct_', 'ak_'), allowanceAmount);
       },
+      async getContractTokenPairs(
+        { rootState: { sdk }, state: { availableTokens } },
+        contractAddress,
+      ) {
+        try {
+          const tokenContract = await sdk.getContractInstance({
+            source: routerInterface,
+            contractAddress,
+          });
+
+          const token0 = await tokenContract.methods.token0();
+          const token1 = await tokenContract.methods.token1();
+
+          const tokens = Object.values(availableTokens);
+
+          return [
+            tokens.find(({ contract }) => contract === token0.decodedResult),
+            tokens.find(({ contract }) => contract === token1.decodedResult),
+          ];
```

Comment:
could we change `.find` to `.filter`?
```suggestion
          return tokens.filter(({ contract }) => [token0.decodedResult, token1.decodedResult].includes(contract));
```

Reply:
we cannot rely on `.filter` because we need the tokens in proper order in the array `[token0, token1]`